fvm/src/kernel/default.rs

```diff
@@ -746,79 +743,60 @@ where
     }
 }
 
-fn draw_randomness(
-    rbase: &[u8; RANDOMNESS_LENGTH],
-    pers: i64,
-    round: ChainEpoch,
-    entropy: &[u8],
-) -> anyhow::Result<[u8; RANDOMNESS_LENGTH]> {
-    let mut state = Params::new().hash_length(32).to_state();
-    state.write_i64::<byteorder::BigEndian>(pers)?;
-    state.write_all(rbase)?;
-    state.write_i64::<byteorder::BigEndian>(round)?;
-    state.write_all(entropy)?;
-    state
-        .finalize()
-        .as_bytes()
-        .try_into()
-        .map_err(anyhow::Error::from)
-}
-
 impl<C> RandomnessOps for DefaultKernel<C>
 where
     C: CallManager,
 {
     fn get_randomness_from_tickets(
         &self,
-        personalization: i64,
         rand_epoch: ChainEpoch,
-        entropy: &[u8],
     ) -> Result<[u8; RANDOMNESS_LENGTH]> {
+        let current_epoch = self.call_manager.context().epoch;
+        if rand_epoch > current_epoch {
```

Comment:
FYI... it _is_ possible to write this as:

```suggestion
        let lookback = current_epoch.checked_sub(rand_epoch)
            .ok_or_else(syscall_error!(IllegalArgument; "randomness epoch {} is in the future", rand_epoch))?;
```

Reply:
(lets you skip the subtraction below)

Reply:
Not that you necessarily should do it this way. I just figured I'd let you know.